docs/managing-dragonfly/operator/dragonfly-configuration.md

```diff
@@ -0,0 +1,24 @@
+---
+sidebar_position: 2
+---
+
+# Dragonfly CRD
+
+Dragonfly Operator uses a dedicated Dragonfly CRD to create and manage dragonfly
+resources. The CRD allows various configurations to define the behaviour of dragonfly
+controller and the dragonfly pods. Below is the table of Dragonfly CRD fields.
+
+| fields | type | Description |
+| ------ | ---- | ----------- |
+| `affinity` | object | Dragonfly pod affinity (Optional)<br/><pre>spec:<br/>  affinity: <br/>    nodeaffinity:<br/>      ...</pre> You can learn more about affinity [here](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity).|
+| `replicas` | int | The total number of Dragonfly instances including the master. |
+| `image` | string | The dragonfly image to use. Default is `docker.dragonflydb.io/dragonflydb/dragonfly:v1.10.0` |
+| `args` | []string | (Optional) Dragonfly container args to pass to the container. Refer to the Dragonfly documentation for the list of supported args. Example - <br/><pre>spec:<br/>  args:<br/>   - "--requirepass=supersecret"</pre> |
+| `annotations` | object | (Optional) Annotations to add to the Dragonfly pods. See [Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) to know more about annotations. |
+| `env` | array | Environmental Variables to add to Dragonfly pods. Example - <br/><pre>spec:<br/>  env:<br/>   - name: DEBUG<br/>     value: true</pre>|
+| `resources` | object | (Optional) Dragonfly container resource limits. Any container limit can be specified.|
+| `tolerations` | object | (Optional) Dragonfly pod tolerations. See [k8s doc](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) to know more about tolerations |
+| `serviceAccountName` | string | (Optional) Dragonfly pod service account name |
+| `authentication` | object | (Optional) Dragonfly Authentication mechanism |
```

Comment:
For nested fields, maybe expand each field as a separate item in the table i.e `authenticaition.passwordFromSecret` like that?